recipes/lbfgspp/meta.yaml

```
@@ -0,0 +1,42 @@
+{% set name = "lbfgspp" %}
+{% set version = "0.2.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/yixuan/lbfgspp/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: 7101744a538c3aff52e10c82267305847b0b5e9d39f9974b4b29812cd1398ff9
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+  host:
+    - eigen
+
```

```suggestion
  host:
    - eigen =3.0.*
  run:
    - eigen =3.0.*
```
Two things:

1. Upstream [explicitly requires eigen 3.0](https://github.com/yixuan/LBFGSpp/blob/00f0f1bd96689bd40e6c8b16cc135665eed5acc9/CMakeLists.txt#L41)
2. To be nice to users, we add eigen to the run section so when they add this package to requirements, eigen is also added (it's mentioned in the headers).

---

I think the cmake version requirements is actually equivalent to `>=3.0,<4`. Otherwise, the builds would have failed already.